whack-a-mole/script.js

```diff
@@ -0,0 +1,159 @@
+(function startFunction(){
+const holes = document.querySelectorAll('.hole');
+const moles = document.querySelectorAll('.mole');
+const scoreScreen = document.getElementById('score');
+const startButton = document.getElementById('start-button');
+const lavelChange = document.getElementById('level-change');
+const minMax = document.getElementById('first-level');
+const gameContinue = document.querySelector('.game-continue');
+const buttons = document.querySelectorAll('.button');
+let gameDuration = 60000;
+let timeIsOver = false;
+let score;
+let startTime = 0;//time of the game beginning
+let timePassed=0;
+let lastHoleNumber;
+
+let min = minMax.dataset.min;
+let max = minMax.dataset.max;
+
+
+
+// get local storage score
+const getScore = () => {
+        scoreScreen.textContent = localStorage.getItem('score');
+        timePassed = localStorage.getItem('timePassed');
+}
+
+
+// set local storage score
+const setScore = () =>{
+           localStorage.setItem('score', scoreScreen.textContent);
+           localStorage.setItem('timePassed', timePassed);
+   }
+
+
+
+
+// Function to get random mole showing up time
+function getShowUpTime(min, max){
+    return Math.round(Math.random()*(max-min)+min);
+}
+
+
+
+
+// Function to get random hole number
+const getHoleNumber = holes => {
+    const randomIdHole = Math.floor(Math.random()*holes.length);
+    const holeNumber = holes[randomIdHole];
+    if(lastHoleNumber === holeNumber){
+        getHoleNumber(holes);
+    }
+    lastHoleNumber = holeNumber;
+    return holeNumber;
+}
+
+
+
+// Function to show the mole
+function showMole(){
+    const time = getShowUpTime(min, max);
+    const hole = getHoleNumber(holes);
+    hole.classList.add('up');
+    timePassed1 = new Date();
+    timePassed2 = timePassed1.getTime();
+    timePassed = timePassed2- startTime;
```

Comment:
```suggestion
    timePassed = new Date().getTime() - startTime;
```
The variables above are declared without let or const. That means they are in the global scope, which is not good